xaitk_saliency/utils/gen_coco_sal.py

```diff
@@ -0,0 +1,262 @@
+        detection id key values, taken from the kwcoco dataset. Each map has a
+        shape of [imgHeight, imgWidth], reffering to the detection's
+        corresponding image.
+
+        :param dets_dset: ``kwcoco`` dataset object containing detections to
+            generate saliency maps for.
+        :param detector: Implementation of ``DetectImageObjects`` to generate
+            detections with.
+        :param img_perturber: Implementation of ``PerturbImage`` to use.
+        :param sal_generator: Implementation of
+            ``GenerateDetectorProposalSaliency`` to use.
+        :param verbose: Option to print out progress statements. Default is
+            false.
+
+        :returns: Dictionary of saliency maps. Keys of this dictionary are the
+            ids of the corresponding detections taken from the ``kwcoco``
+            dataset. Each saliency map has shape [imgHeight, imgWidth],
+            reffering to the detection's corresponding image.
+        """
+
+        if verbose:
+            _log: Callable[[str], None] = LOG.info
+        else:
+            _log = _mute_log
+
+        num_imgs = len(dets_dset.imgs)
+
+        sal_maps = {}
+        for img_i, (img_id, det_ids) in enumerate(dets_dset.gid_to_aids.items()):
+
+            num_dets = len(det_ids)
+            # continue if there are no dets for this image
+            if num_dets == 0:
+                continue
+
+            img_file = dets_dset.get_image_fpath(img_id)
+            ref_img = np.asarray(Image.open(img_file))
+
+            ref_dets_mat = _coco_dets_to_formatted_mat(dets_dset, det_ids)
+
+            _log(f'[{img_id}] ({img_i+1}/{num_imgs}) {num_dets} reference detections found.')
+
+            _log(f'[{img_id}] ({img_i+1}/{num_imgs}) Perturbing image.')
+
+            pert_masks = img_perturber(ref_img)
+            pert_imgs = occlude_image_batch(ref_img, pert_masks)
+
+            _log(f'[{img_id}] ({img_i+1}/{num_imgs}) Getting perturbation detections.')
+
+            pert_dets_iter = detector(pert_imgs)
+
+            _log(f'[{img_id}] ({img_i+1}/{num_imgs}) Generating saliency maps.')
+
+            pert_dets_mat = _dets_to_formatted_mat(pert_dets_iter, dets_dset.cats)
+
+            img_sal_maps = sal_generator(ref_dets_mat, pert_dets_mat, pert_masks)
+
+            for i, det_id in enumerate(det_ids):
+                sal_maps[det_id] = img_sal_maps[i]
+
+        return sal_maps
+
+    def _name_to_cat_id(
+        name: str,
+        cats: dict
+    ) -> int:
+        """
+        Converts category name to id.
+
+        :param name: Category name.
+        :param cats: Dictionary of categories, as taken from the "categories"
+            section of a COCO style annotation dictionary.
+        :returns: Id of matching category, or -1 if no matching category is
+            found.
+        """
+        for cat_id in cats:
+            if name == cats[cat_id]['name']:
+                return cat_id
```

Comment:
Accesses (read: "expensive" function calls) into the `cats` dict could be reduced a little by using:
```suggestion
        for cat_id, cat_obj in cats.items():
            if name == cat_obj['name']:
                return cat_id
```